lib.nix

```diff
@@ -10,6 +10,20 @@
 
   isInStore = lib.hasPrefix builtins.storeDir;
 
+  valueWithContext = {
+    create = value: context: {inherit value context;};
```

Comment:
Very small nitpick, but I feel like `new` is more common as a constructor name. Very unimportant, though

Reply:
@yannham Do you still have some review you want to share? If so, could you?